include/circt/Dialect/HW/HWPasses.h

```diff
@@ -15,8 +15,8 @@
 
 #include "mlir/Pass/Pass.h"
 #include "mlir/Pass/PassRegistry.h"
-#include "llvm/ADT/Optional.h"
 #include <memory>
+#include <optional>
```

Comment:
This seems like it doesn't belong here and should be defined in the passes that need it. Out-of-scope for this PR.